source/extensionPoints/__init__.py

```diff
@@ -66,22 +70,28 @@ def notifyOnce(self, **kwargs):
 				log.exception(f"Error running handler {handler} for {self}. Exception {e}")
 
 
-FilterValueTypeT = TypeVar("FilterOutputTypeT")
+FilterValueTypeT = TypeVar("FilterValueTypeT")
 
 
-class Filter(HandlerRegistrar, Generic[FilterValueTypeT]):
+# Typing doesn't allow us to typehint a Callable with one required argument and several kwargs.
+# Using a typing_extensions.Protocol for this results in false negatives.
+class Filter(
+		HandlerRegistrar[Union[Callable[..., FilterValueTypeT], Callable[[FilterValueTypeT], FilterValueTypeT]]],

```

Comment:
could this type be refactored into a named variable or multiple named variables? it would improve legibility

Reply:
I tried that initially, but then at least VS Code was unable to infer the expected types when calling register on the action. May be that's just VSCode specific?

Reply:
I think this is fine to leave as is for this case.
In my experience, the infer was fine (i.e. method discovery on a class), just that the type hint on focus/hover used the named types rather than unfurling.
I'm not sure if it is intended behaviour or not, e.g. a developer may generally want to use the defined type, rather than a hard to comprehend unfurled type.

Reply:
I also assume it should be allowed to store this in a variable, but I think the problem is as follows. When defining a generic, the type checker inspects the TypeVars that can be found directly in the generic definition and doesn't inspect variables that reference typevars.
I could file an issue against Pyright for this?

Reply:
I'm not sure if `TypeVar` syntax is required here.
What I pictured was this, but is this syntax problematic also?

```py
_FilterFunctionT = Union[Callable[..., FilterValueTypeT], Callable[[FilterValueTypeT], FilterValueTypeT]]
class Filter(HandlerRegistrar[_FilterFunctionT]): pass
```

Reply:
This is exactly what I wanted to do. The problem is that you're passing a variable containing a Callable with a typevar to a Generic (i.e. HandlerRegistrar inherrits from Generic), and then at least PyRight seems to be unable to infer the right signature when calling register on a resulting filter.
Have you tried this in PyCharm, and what does PyCharm report when you call register on braille.filter_displaySize? Does it tell you that it requires a function returning an int, or does it tell you that it requires a _FilterFunctionT?
If the former, it's clearly VS Code that's in error here and I'm happy to change this.

Reply:
Thanks for investigating. I have not tried in this PyCharm, I'm also a VS code user.
I think this is find to leave as is.